sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -81,15 +83,31 @@ class HiveContext private[hive](
     sc: SparkContext,
     cacheManager: CacheManager,
     listener: SQLListener,
-    @transient private val execHive: HiveClientImpl,
-    @transient private val metaHive: HiveClient,
-    isRootContext: Boolean)
-  extends SQLContext(sc, cacheManager, listener, isRootContext) with Logging {
+    @transient protected[hive] val executionHive: HiveClientImpl,
```

Comment:
We have not changed temp function part. executionHive is still used when we look up Hive UDFs. The next step is to remove the usage of executionHive when we handle Hive UDFs. Then, we should be able to remove executionHive (So, HiveClient is only used to talk to real metastores.).